code/modules/holodeck/computer.dm

```diff
@@ -110,63 +121,277 @@
 	. = TRUE
 	switch(action)
 		if("load_program")
-			var/program_to_load = text2path(params["type"])
-			if(!ispath(program_to_load))
-				return FALSE
-			var/valid = FALSE
-			var/list/checked = program_cache.Copy()
-			if(obj_flags & EMAGGED)
-				checked |= emag_programs
-			for(var/prog in checked)
-				var/list/P = prog
-				if(P["type"] == program_to_load)
-					valid = TRUE
-					break
-			if(!valid)
-				return FALSE
-
-			var/area/A = locate(program_to_load) in GLOB.sortedAreas
-			if(A)
-				load_program(A)
+			var/program_to_load = params["id"]
+			//load the map_template that program_to_load represents
+			if(program_to_load)
+				load_program(program_to_load)
 		if("safety")
 			if((obj_flags & EMAGGED) && program)
 				emergency_shutdown()
 			nerf(obj_flags & EMAGGED)
 			obj_flags ^= EMAGGED
 			say("Safeties restored. Restarting...")
 
+/*
+	basically the equivalent of map_template/load except it calls parsed.load with placeOnTop = FALSE (so holodeck programs dont stack in the baseturf list)
+	and passes spawned_atoms from parsed to the holodeck template datum
+*/
+/datum/map_template/holodeck/load(turf/target_turf, centered = FALSE, )
+	if(centered)
+		target_turf = locate(target_turf.x - round(width/2) , target_turf.y - round(height/2) , target_turf.z)
+	if(!target_turf)
+		return
+	if(target_turf.x+width > world.maxx)
+		return
+	if(target_turf.y+height > world.maxy)
+		return
+
+	var/list/border = block(locate(max(target_turf.x-1, 1),	max(target_turf.y-1, 1), target_turf.z),
+							locate(min(target_turf.x+width+1, world.maxx),	min(target_turf.y+height+1, world.maxy), target_turf.z))
+	for(var/_turf in border)
+		var/turf/turf_to_disable = _turf
+		SSair.remove_from_active(turf_to_disable) //stop processing turfs along the border to prevent runtimes, we return it in holodeckTemplateBounds()
+		turf_to_disable.atmos_adjacent_turfs?.Cut()
+
+	// Accept cached maps, but don't save them automatically - we don't want
+	// ruins clogging up memory for the whole round.
+	var/datum/parsed_map/parsed = cached_map || new(file(mappath))
+	cached_map = keep_cached_map ? parsed : null
+	if(!parsed.load(target_turf.x, target_turf.y, target_turf.z, cropMap=TRUE, no_changeturf=(SSatoms.initialized == INITIALIZATION_INSSATOMS), placeOnTop=FALSE))
+		return
+	var/list/bounds = parsed.bounds
+	if(!bounds)
+		return
+
+	if(!SSmapping.loading_ruins) //Will be done manually during mapping ss init
+		repopulate_sorted_areas()
+
+	//initialize things that are normally initialized after map load
+
+	spawned_atoms = parsed.holodeckTemplateBounds()
+
+	for (var/obj/base_container in spawned_atoms)//this didnt seem to work when i put this in holodeckTemplateBounds
```

Comment:
Is this necessary if you already iterate them one by one during template initialization.

Reply:
this is for lockers that create other items on initialization, thus the insides arent actually spawned from the template which makes them untracked otherwise

Reply:
ill probably end up putting this part into load_program

Reply:
leaving this open because its possible theres a better way to accomplish this without manually adding all contents of every spawned container just in case they create items on Initialize()

Reply:
as of right now this is fixed unless i get smacked for altering ssatoms to return a list of atoms created from a template (if the template is set to return a list, which only holodeck templates are currently)